analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -282,8 +282,23 @@ node switch
         public static bool HasConstantValue(this ExpressionSyntax expression, SemanticModel semanticModel) =>
             expression.RemoveParentheses().IsAnyKind(LiteralSyntaxKinds) || expression.FindConstantValue(semanticModel) != null;
 
-        public static string GetStringValue(this SyntaxNode node, SemanticModel semanticModel) =>
-            CsharpStringValueHelper.Instance.GetStringValue(node, semanticModel);
+        public static string StringValue(this SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (node != null)
+            {
+                if (node.IsAnyKind(SyntaxKind.StringLiteralExpression, SyntaxKindEx.Utf8StringLiteralExpression)
+                    && node is LiteralExpressionSyntax literal)
+                {
+                    return literal.Token.ValueText;
+                }
+                else if (node is InterpolatedStringExpressionSyntax interpolatedExpression)
+                {
+                    interpolatedExpression.TryGetGetInterpolatedTextValue(semanticModel, out var interpolatedValue);
+                    return interpolatedValue ?? interpolatedExpression.GetContentsText() ?? null;
```

Comment:
This logic is not clean. The `bool` result of the `TryGetGetGetGetStutter` is ignored and the after-effect of nullability is used instead. The `TryGet..` should be used as ternary instead.

Topic for another PR would be to remove the `TryGet`+`out` logic and return `interpolatedValue` directly instead to clean that part. Or maybe in this PR, I won't mind.

Reply:
I would say it's for another PR - as it's used in quite some places.
I'll open an issue and refactor after vacations - should not be super crazy.